### PR TITLE
Improve hg support

### PR DESCRIPTION
**Pull request by andrewshadura**
Hi,

These are a couple of changes which are supposed to improve the state of Mercurial support in Weblate. I haven’t tested them yet, but I will and I also hope I will come with some real tests to test this functionality in more or less real situations, especially those when it’s failing for me at the moment.

Thanks for considering this PR.

### CODE REVIEW

**Comment by nijel** [comment]
Does any of the changes depend on some particular mercurial version? The tests just hang on Travis CI which has 2.0.2. It works fine on 4.0 I have locally installed.

**Comment by nijel** [comment]
Hmm, actually Travis has both 2.0.2 (which should not be supported by Weblate at all) and 4.0.1 installed from pip, I've fixed this mess in 371a0d02e2ac420c67c7404d3a2a7a7963b936e7.

File: weblate/trans/vcs.py [hunk 1]
```diff
@@ -991,6 +991,25 @@ def check_config(self):
...
+
+        merge_driver = os.path.abspath(
+            os.path.join(
+                os.path.dirname(os.path.dirname(os.path.dirname(__file__))),
+                'examples',
+                'git-merge-gettext-po'
+            )
+        )
+
+        if os.path.exists(merge_driver):
```
Comment (nijel) [inline weblate/trans/vcs.py]: I've factored out this part from Git in 816e3f270, can you please update your code to use new method as well?

**Comment by nijel** [comment]
The build still hangs on Travis, so it's probably not really working as expected...

**Comment by nijel** [comment]
The documentation is here: https://docs.weblate.org/en/latest/contributing.html#testsuite

The particular problematic test (hanging on Travis CI) can be executed by:

```
./manage.py test --settings weblate.settings_test weblate.trans.tests.test_remote.MercurialMultiRepoTest
```

It really looks to me like the `ui.merge` configuration didn't work and it's interactively waiting for input.

**Comment by andrewshadura** [comment]
Hmm, it still fails on Travis, even though it succeeded locally.

**Comment by andrewshadura** [comment]
G
​reat, thanks!

Can you please also update the hosted instance so that I can test it in
action? I
​t's been in a very confused state since about middle of Se​ptember.

-- 
Cheers,
  Andrew

**Comment by nijel** [comment]
I've just updated it :-)

**Pull request merged**